tests/unit/amo/components/TestHeader.js

```diff
@@ -40,4 +59,35 @@ describe(__filename, () => {
     expect(root.find('.Header-title').prop('children'))
       .toContain('Firefox Add-ons');
   });
+
+  it('displays `login` text when user is not signed in', () => {
+    const wrapper = renderHeader();
+
+    expect(wrapper.find(AuthenticateButton)).toHaveLength(1);
+    expect(wrapper.find(DropdownMenu)).toHaveLength(0);
+  });
+
+  it('displays a menu and the username when user is signed in', () => {
+    const { store } = dispatchSignInActions({ username: 'babar' });
+    const wrapper = renderHeader({ store });
+
+    expect(wrapper.find(DropdownMenu)).toHaveLength(1);
+    expect(wrapper.find(DropdownMenu)).toHaveProp('text', 'babar');
+  });
+
+  it('allows a signed-in user to log out', () => {
+    const { store } = dispatchSignInActions({ username: 'babar' });
+    const wrapper = renderHeader({ store });
+    const mockApi = sinon.mock(api);
+
+    mockApi
+      .expects('logOutFromServer')
```

Comment:
This component is relying on `props.handleLogOut()` so you should be mocking the expectation for how `handleLogOut()` is called. As I mentioned in a comment, the `createHandleLogOutFunction` needs tests of its own which will look like the one here -- i.e. `mockApi.expects('logOutFromServer')`. 

This kind of approach allows us to separate the responsibilities of each unit that makes up the application.

In order to inject a `sinon.stub()` for `props.handleLogOut` you may need to change [this line](https://github.com/willdurand/addons-frontend/blob/dfa71a8f67fd9322146114e65e1ed778c086d4ff/src/amo/components/Header/index.js#L137) but I'm not sure:

```js
export const mapDispatchToProps = (dispatch, ownProps) => ({
  handleLogOut: ownProps.handleLogOut || createHandleLogOutFunction(dispatch),
});
```